sdk/core/typespec_client_core/src/http/policies/retry/exponential.rs

```diff
@@ -18,6 +21,7 @@ pub(crate) struct ExponentialRetryPolicy {
     max_elapsed: Duration,
     max_delay: Duration,
     retry_headers: RetryHeaders,
+    status_codes: Vec<StatusCode>,
```

Comment:
Nit: I'd call these `retry_status_codes` like you do in the `PipelineOptions`. Helps more easily identity them. Plus, makes it sound more like input. I was also thinking that for the parameter names, but they do get a bit long that way. Something to think about.